src/coreclr/scripts/superpmi.py

```diff
@@ -1080,12 +1085,12 @@ async def run_crossgen2(print_prefix, assembly, self):
                 old_env = os.environ.copy()
                 os.environ.update(crossgen2_command_env)
 
-                # Note: crossgen2 compiles in parallel, so we don't need to use per-assembly parallelism here.
-                # If we ever want to use parallelism in this script instead, remove the "subproc_count=1"
-                # argument to the AsyncSubprocessHelper constructor below, and pass "--parallelism:1" to
-                # the crossgen2 process, above.
-                # We still use the same code pattern here as for PMI and crossgen1 collections, for consistency.
-                helper = AsyncSubprocessHelper(assemblies, subproc_count=1, verbose=True)
+                # Note: crossgen2 compiles in parallel by default. However, it seems to lead to sharing violations
+                # in SuperPMI collection, accessing the MC file. So, disable crossgen2 parallism by using
+                # the "--parallelism:1" switch, and allowing coarse-grained (per-assembly) parallelism here.
+                # It turns out this works better anyway, as there is a lot of non-parallel time between
+                # crossgen2 parallel compilations.
+                helper = AsyncSubprocessHelper(assemblies, verbose=True)
```

Comment:
Seems a bit odd to me that SPMI can handle concurrent jit requests from the runtime but not from crossgen2, but if it's faster to run this way I guess there's not much to be gained by a deeper look.

Reply:
It would be useful to understand why the failures happen. But there are bigger problems to solve...